app/src/main/java/com/edu/uni/augsburg/uniatron/ui/MainActivity.java

```diff
@@ -0,0 +1,134 @@
+ * The main activity is the entry point of the app.
+ *
+ * @author Fabio Hellmann
+ */
+public class MainActivity extends AppCompatActivity implements ViewPager.OnPageChangeListener {
+
+    private static final int NAV_POSITION_HISTORY = 0;
+    private static final int NAV_POSITION_HOME = 1;
+    private static final int NAV_POSITION_SETTING = 2;
+
+    @BindView(R.id.viewPager)
+    ViewPager mViewPager;
+    @BindView(R.id.navigation)
+    BottomNavigationView mNavigation;
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        ButterKnife.bind(this);
+
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().hide();
+        }
+
+        final ScreenSlidePagerAdapter mScreenSlideAdapter =
+                new ScreenSlidePagerAdapter(getSupportFragmentManager());
+        mViewPager.setAdapter(mScreenSlideAdapter);
+        mViewPager.addOnPageChangeListener(this);
+
+        mNavigation.setOnNavigationItemSelectedListener(item -> {
+            switch (item.getItemId()) {
+                case R.id.navigation_home:
+                    mViewPager.setCurrentItem(NAV_POSITION_HOME);
+                    return true;
+                case R.id.navigation_history:
+                    mViewPager.setCurrentItem(NAV_POSITION_HISTORY);
+                    return true;
+                case R.id.navigation_settings:
+                    mViewPager.setCurrentItem(NAV_POSITION_SETTING);
+                    return true;
+                default:
+                    return false;
+            }
+        });
+        mNavigation.setSelectedItemId(R.id.navigation_home);
+    }
+
+    @Override
+    public void onPageScrolled(final int position,
+                               final float positionOffset,
+                               final int positionOffsetPixels) {
+        // can be ignored
+    }
+
+    @Override
+    public void onPageSelected(final int position) {
+        switch (position) {
+            case NAV_POSITION_HISTORY:
+                mNavigation.setSelectedItemId(R.id.navigation_history);
+                break;
+            case NAV_POSITION_SETTING:
+                mNavigation.setSelectedItemId(R.id.navigation_settings);
+                break;
+            case NAV_POSITION_HOME:
+            default:
+                mNavigation.setSelectedItemId(R.id.navigation_home);
+                break;
+        }
+    }
+
+    @Override
+    public void onPageScrollStateChanged(final int state) {
+        // can be ignored
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (mViewPager.getCurrentItem() == NAV_POSITION_HOME) {
+            // If the user is currently looking at the home screen,
+            // allow the system to handle the Back button. This
+            // calls finish() on this activity and pops the back stack.
+            super.onBackPressed();
```

Comment:
Could implement a 'press again to exit' toast on the home screen, to prevent accidental closing of the app. This helps the user, as the button functionality changes throughout the app

Reply:
This behaviour is used in serveral other apps as well. A toast would be displayed while the screen is moved to the home fragment. Which, I think, is distructing from the animation/navigation behaviour.